test/markup/java/numbers.txt

```diff
@@ -7,3 +7,24 @@ long maxLong = 0x7fff_ffff_ffff_ffffL;
 byte nybbles = 0b0010_0101;
 long bytes = 0b11010010_01101001_10010100_10010010;
 int n = 1234 + Contacts._ID;
+float f = 0x1.4p2f;
+double d = 0x.ep-6;
+int octal = 0777;
+float f = 2e3f;
+double d = 1.2e4D;
+a = 0x4fa6p2;
+b = 0x.4p2;
+c = 0xa.ffp3f;
+d = 0x1.0p2F;
+e = 0x1.0p2f;
+f = 0x1p1;
+g = 0x.3p4d;
+h = 0x1.2ep5D;
+i = 0x1.p2;
+int i = 23;
+byte mask = 0x0f;
+int i = 4;
+byte mask = 0xa;
+float f = 5.4;
+float f = 2e3;
+int n = 0b1;
```

Comment:
Are things like:

```
float f = .2;
float f = 3.;
```

...allowed in Java?

Reply:
yes it is, you're right